sdk/servicebus/Azure.Messaging.ServiceBus/src/RuleManager/ServiceBusRuleManager.cs

```diff
@@ -0,0 +1,278 @@
+        /// You can add multiple rules with distinct names to the same subscription.
+        /// Multiple filters combine with each other using logical OR condition. i.e., If any filter succeeds, the message is passed on to the subscription.
+        /// </remarks>
+        ///
+        /// <returns>A task instance that represents the asynchronous add rule operation.</returns>
+        public virtual async Task AddRuleAsync(
+            CreateRuleOptions options,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotDisposed(IsClosed, nameof(ServiceBusRuleManager));
+            Argument.AssertNotNull(options, nameof(options));
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            EntityNameFormatter.CheckValidRuleName(options.Name);
+            ServiceBusEventSource.Log.AddRuleStart(Identifier, options.Name);
+
+            try
+            {
+                await InnerRuleManager.AddRuleAsync(
+                    new RuleProperties(options),
+                    cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception exception)
+            {
+                ServiceBusEventSource.Log.AddRuleException(Identifier, exception.ToString());
+                throw;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            ServiceBusEventSource.Log.AddRuleComplete(Identifier);
+        }
+
+        /// <summary>
+        /// Removes the rule on the subscription identified by <paramref name="ruleName" />.
+        /// </summary>
+        ///
+        /// <param name="ruleName">Name of the rule</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>A task instance that represents the asynchronous remove rule operation.</returns>
+        public virtual async Task RemoveRuleAsync(
+            string ruleName,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotDisposed(IsClosed, nameof(ServiceBusRuleManager));
+            Argument.AssertNotNullOrEmpty(ruleName, nameof(ruleName));
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+            ServiceBusEventSource.Log.RemoveRuleStart(Identifier, ruleName);
+
+            try
+            {
+                await InnerRuleManager.RemoveRuleAsync(
+                    ruleName,
+                    cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception exception)
+            {
+                ServiceBusEventSource.Log.RemoveRuleException(Identifier, exception.ToString());
+                throw;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
```

Comment:
Do we really want to check cancellation here?   The operation is already complete.

If we do decide to keep it, if this line throws, you'll get a "Remove Rule Start" log event without a matching error/complete.   Is that intended?